### PR TITLE
feat: implemented aggregate functions

### PR DESCRIPTION
**Pull request by michaellee8**

### CODE REVIEW

**Review by phuctm97 — CHANGES_REQUESTED** [review]
Please run a code format using Android Studio to properly format the code to have consistent style: Right click on the project -> `Reformat Code` -> Select reformat all, rearrange entries, optimize imports

**Comment by michaellee8** [comment]
reformatted

File: lib/torus_direct.dart [hunk 1]
```diff
@@ -18,26 +19,72 @@ enum TorusLogin {
   jwt
 }
 
+enum AggregateVerifierType { single_id_verifier }
```
Comment (phuctm97) [inline lib/torus_direct.dart:22]: I think we should use camel case, most Flutter/Dark official examples use camel case values in enum. This makes a better DX.

File: ios/Classes/SwiftTorusDirectPlugin.swift [hunk 2]
```diff
@@ -9,134 +9,239 @@ struct TorusDirectArgs {
...
+                        code: "MissingArgumentException",
+                        message: "Missing triggerLogin arguments",
+                        details: nil))
+                return
+            }
+            guard let loginProvider = LoginProviders(rawValue: typeOfLogin) else {
+                result(FlutterError(
+                        code: "InvalidTypeOfLogin",
+                        message: "Invalid type of login",
+                        details: nil))
```
Comment (phuctm97) [inline ios/Classes/SwiftTorusDirectPlugin.swift:72]: All error codes are ending with `..Exception`, I think this should share the same pattern too.

File: ios/Classes/SwiftTorusDirectPlugin.swift [hunk 3]
```diff
@@ -9,134 +9,239 @@ struct TorusDirectArgs {
...
+            for details in subVerifierDetailsArray {
+                guard let loginProvider = LoginProviders(
+                    rawValue: details["typeOfLogin"] as! String
+                ) else {
+                    result(FlutterError(
+                            code: "InvalidTypeOfLogin",
+                            message: "Invalid type of login",
+                            details: nil))
+                    return
+                }
```
Comment (phuctm97) [inline ios/Classes/SwiftTorusDirectPlugin.swift:127]: All error codes are ending with ..Exception, I think this should share the same pattern too.

**Review by phuctm97 — CHANGES_REQUESTED** [review]

File: lib/torus_direct.dart (hunk 1 above)
Comment (michaellee8) [reply lib/torus_direct.dart:22]: I intends to match the pattern used in other sdks here, also does not really seems to support assigning a string value to a enum.

File: lib/torus_direct.dart [hunk 4]
```diff
@@ -18,26 +19,72 @@ enum TorusLogin {
...
+  final String verifier;
+  final String clientId;
+  final Map jwtParams;
+
+  SubVerifierDetails({
+    required this.typeOfLogin,
+    required this.verifier,
+    required this.clientId,
+    this.jwtParams = const {},
+  });
```
[the remark this answers is not in the document]
Comment (michaellee8) [reply lib/torus_direct.dart:45]: refactored

File: ios/Classes/SwiftTorusDirectPlugin.swift (hunk 3 above)
Comment (michaellee8) [reply ios/Classes/SwiftTorusDirectPlugin.swift]: refactored

**Comment by michaellee8** [comment]
@phuctm97 sorry for the late reply here, please see if it looks good to u

**Review by phuctm97 — APPROVED** [review]
lgtm, please resolve conflicts and test again.

**Review by phuctm97 — APPROVED** [review]
lgtm

**Pull request merged**